### DISCUSSION TITLE
I want to improve Force-Push.

### DISCUSSION THREAD

**Discussion by 911432** [community/community#130410] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

If the same person did Force-Push, I would like to leave only the last part, not the intermediate process.
![force-pushed](https://github.com/community/community/assets/30204381/2dd37272-77f0-4131-9c49-a6d65d774d58)

**Comment by nandodev-net** [comment]
Hello @911432 , to achieve a cleaner commit history, where only the last part of the force-push is left and not the intermediate processes, you can use `git rebase` or `git reset`. Below are the steps to accomplish this:

1. **Backup your current branch**: It's always a good practice to have a backup before performing destructive operations in Git.

    ```bash
    git checkout -b backup-branch
    ```

2. **Interactive rebase**: Use interactive rebase to edit the commit history.

    ```bash
    git checkout main
    git rebase -i HEAD~N  # N is the number of commits back you want to rewrite
    ```

3. **Edit the interactive rebase**: In the editor that opens, change `pick` to `squash` (or `s`) for the intermediate commits you want to combine into the final commit.

    ```text
    pick e93f3ee Initial commit
    squash 29c99ca Some intermediate commit
    squash 12b5f46 Another intermediate commit
    pick eb21ae3 Final commit
    ```

4. **Update the commit history**: Save and close the editor. Git will combine the intermediate commits into one. You might be prompted to edit the commit message for the combined commit.

5. **Force push the rewritten history**: If you have already pushed these commits, you will need to force push to overwrite the history on the remote repository.

    ```bash
    git push origin main --force
    ```

This process will leave only the final commit visible in the history, removing the intermediate commits made by the same person.

**Important Note**: Force pushing (`--force`) can overwrite the history in the remote repository, which might cause issues for other collaborators. Make sure to communicate any changes to your team.

**Reply by 911432** [reply]
it doesn't work
![force-pushed](https://github.com/community/community/assets/30204381/4296d972-45ae-43bd-85af-d8e358b5b6e9)
